internal/elasticsearch/enrich/resource.go

```diff
@@ -0,0 +1,129 @@
+	resp.Schema = GetResourceSchema()
+}
+
+func GetResourceSchema() schema.Schema {
+	return schema.Schema{
+		MarkdownDescription: "Managing Elasticsearch enrich policies. See: https://www.elastic.co/guide/en/elasticsearch/reference/current/enrich-apis.html",
+		Blocks: map[string]schema.Block{
+			"elasticsearch_connection": providerschema.GetEsFWConnectionBlock("elasticsearch_connection", false),
+		},
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "Internal identifier of the resource",
+				Computed:            true,
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Name of the enrich policy to manage.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(1, 255),
+				},
+			},
+			"policy_type": schema.StringAttribute{
+				MarkdownDescription: "The type of enrich policy, can be one of geo_match, match, range.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf("geo_match", "match", "range"),
+				},
+			},
+			"indices": schema.ListAttribute{
+				MarkdownDescription: "Array of one or more source indices used to create the enrich index.",
+				ElementType:         types.StringType,
+				Required:            true,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.List{
+					listvalidator.SizeAtLeast(1),
+				},
+			},
+			"match_field": schema.StringAttribute{
+				MarkdownDescription: "Field in source indices used to match incoming documents.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(1, 255),
+				},
+			},
+			"enrich_fields": schema.ListAttribute{
+				MarkdownDescription: "Fields to add to matching incoming documents. These fields must be present in the source indices.",
+				ElementType:         types.StringType,
+				Required:            true,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.List{
+					listvalidator.SizeAtLeast(1),
+				},
+			},
+			"query": schema.StringAttribute{
+				MarkdownDescription: "Query used to filter documents in the enrich index. The policy only uses documents matching this query to enrich incoming documents. Defaults to a match_all query.",
+				Optional:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexp.MustCompile(`^\{.*\}$`), "must be valid JSON"),
+				},
```

Comment:
```suggestion
				CustomType:  jsontypes.NormalizedType{},
```

Use the JSON custom type defined in `github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes`